creator-node/src/components/replicaSet/URSMRegistrationComponentService.js

```diff
@@ -33,36 +33,24 @@ const { recoverWallet, signatureHasExpired } = require('../../apiSigning')
  * @param {string} reqSignature signature generated by requesting node, used in public key recovery
  */
 const respondToURSMRequestForSignature = async ({ libs: audiusLibs, nodeConfig }, logger, spID, reqTimestamp, reqSignature) => {
-  if (!spID || !reqTimestamp || !reqSignature) {
-    throw new ErrorBadRequest('Must provide all required query parameters: spID, timestamp, signature')
-  }
-
-  spID = parseInt(spID)
+  const {
+    ownerWalletFromSPFactory,
+    delegateOwnerWalletFromSPFactory,
+    nodeEndpointFromSPFactory,
+    parsedSpId
+  } = getSPInfo({ audiusLibs, spID, reqTimestamp, reqSignature })
 
-  /**
-   * Fetch node info from L1 ServiceProviderFactory for spID
-   */
-  const spRecordFromSPFactory = await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceEndpointInfo(
-    'content-node',
-    spID
-  )
-  let {
-    owner: ownerWalletFromSPFactory,
-    delegateOwnerWallet: delegateOwnerWalletFromSPFactory,
-    endpoint: nodeEndpointFromSPFactory
-  } = spRecordFromSPFactory
-  delegateOwnerWalletFromSPFactory = delegateOwnerWalletFromSPFactory.toLowerCase()
+  spID = parsedSpId
 
-  /**
-   * Reject if node is not registered as valid SP on L1 ServiceProviderFactory
-   */
-  if (
-    LibsUtils.isZeroAddress(ownerWalletFromSPFactory) ||
-    LibsUtils.isZeroAddress(delegateOwnerWalletFromSPFactory) ||
-    !nodeEndpointFromSPFactory
-  ) {
-    throw new ErrorBadRequest(`SpID ${spID} is not registered as valid SP on L1 ServiceProviderFactory`)
-  }
+  await verifyRequesterIsValidSP({
```

Comment:
moved the short circuit delegate owner wallet check is already on ursm after this call